From a5cfdac473291cf476dfb952c7a3a7164bf2a94e Mon Sep 17 00:00:00 2001
From: Vladimir-csp <vladimir-csp@yandex.ru>
Date: Mon, 18 Nov 2019 18:28:02 +0300
Subject: [PATCH 1/2] add aft-mtp-mount support (android-file transfer)

---
 src/ptk/ptk-handler.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ptk/ptk-handler.c b/src/ptk/ptk-handler.c
index 5d00e66f9..fd1264181 100644
--- a/src/ptk/ptk-handler.c
+++ b/src/ptk/ptk-handler.c
@@ -405,9 +405,9 @@ const Handler handlers_net[]=
     {
         "hand_net_+mtp",
         "mtp",
-        "mtp mtab_fs=fuse.jmtpfs mtab_fs=fuse.simple-mtpfs mtab_fs=fuse.mtpfs mtab_fs=fuse.DeviceFs(*",
+        "mtp mtab_fs=aft-mtp-mount mtab_fs=fuse.jmtpfs mtab_fs=fuse.simple-mtpfs mtab_fs=fuse.mtpfs mtab_fs=fuse.DeviceFs(*",
         "",
-        "mtpmount=\"$(which jmtpfs || which simple-mtpfs || which mtpfs || which go-mtpfs)\"\nif [ -z \"$mtpmount\" ]; then\n    echo \"To mount mtp:// you must install jmtpfs, simple-mtpfs, mtpfs, or go-mtpfs,\"\n    echo \"or add a custom protocol handler.\"\n    exit 1\nelif [ \"${mtpmount##*/}\" = \"go-mtpfs\" ]; then\n    # Run go-mtpfs in background, as it does not exit after mount\n    outputtmp=\"$(mktemp --tmpdir spacefm-go-mtpfs-output-XXXXXXXX)\" || exit 1\n    go-mtpfs \"%a\" &> \"$outputtmp\" &\n    sleep 2s\n    [[ -e \"$outputtmp\" ]] && cat \"$outputtmp\" ; rm -f \"$outputtmp\"\n    # set success status only if positive that mountpoint is mountpoint\n    mountpoint \"%a\"\nelse\n    $mtpmount \"%a\"\nfi\n",
+        "mtpmount=\"$(which aft-mtp-mount || which jmtpfs || which simple-mtpfs || which mtpfs || which go-mtpfs)\"\nif [ -z \"$mtpmount\" ]; then\n    echo \"To mount mtp:// you must install jmtpfs, simple-mtpfs, mtpfs, or go-mtpfs,\"\n    echo \"or add a custom protocol handler.\"\n    exit 1\nelif [ \"${mtpmount##*/}\" = \"go-mtpfs\" ]; then\n    # Run go-mtpfs in background, as it does not exit after mount\n    outputtmp=\"$(mktemp --tmpdir spacefm-go-mtpfs-output-XXXXXXXX)\" || exit 1\n    go-mtpfs \"%a\" &> \"$outputtmp\" &\n    sleep 2s\n    [[ -e \"$outputtmp\" ]] && cat \"$outputtmp\" ; rm -f \"$outputtmp\"\n    # set success status only if positive that mountpoint is mountpoint\n    mountpoint \"%a\"\nelif [ \"${mtpmount##*/}\" = \"aft-mtp-mount\" ]; then\n    $mtpmount -f \"%a\"\nelse\n    $mtpmount \"%a\"\nfi\n",
         FALSE,
         "fusermount -u \"%a\"",
         FALSE,

From ac481b889811137d604cc7ef13eddae2eb7d742f Mon Sep 17 00:00:00 2001
From: Vladimir-csp <vladimir-csp@yandex.ru>
Date: Mon, 18 Nov 2019 18:29:51 +0300
Subject: [PATCH 2/2] fuse

---
 src/ptk/ptk-handler.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ptk/ptk-handler.c b/src/ptk/ptk-handler.c
index fd1264181..204cb70de 100644
--- a/src/ptk/ptk-handler.c
+++ b/src/ptk/ptk-handler.c
@@ -405,7 +405,7 @@ const Handler handlers_net[]=
     {
         "hand_net_+mtp",
         "mtp",
-        "mtp mtab_fs=aft-mtp-mount mtab_fs=fuse.jmtpfs mtab_fs=fuse.simple-mtpfs mtab_fs=fuse.mtpfs mtab_fs=fuse.DeviceFs(*",
+        "mtp mtab_fs=fuse.aft-mtp-mount mtab_fs=fuse.jmtpfs mtab_fs=fuse.simple-mtpfs mtab_fs=fuse.mtpfs mtab_fs=fuse.DeviceFs(*",
         "",
         "mtpmount=\"$(which aft-mtp-mount || which jmtpfs || which simple-mtpfs || which mtpfs || which go-mtpfs)\"\nif [ -z \"$mtpmount\" ]; then\n    echo \"To mount mtp:// you must install jmtpfs, simple-mtpfs, mtpfs, or go-mtpfs,\"\n    echo \"or add a custom protocol handler.\"\n    exit 1\nelif [ \"${mtpmount##*/}\" = \"go-mtpfs\" ]; then\n    # Run go-mtpfs in background, as it does not exit after mount\n    outputtmp=\"$(mktemp --tmpdir spacefm-go-mtpfs-output-XXXXXXXX)\" || exit 1\n    go-mtpfs \"%a\" &> \"$outputtmp\" &\n    sleep 2s\n    [[ -e \"$outputtmp\" ]] && cat \"$outputtmp\" ; rm -f \"$outputtmp\"\n    # set success status only if positive that mountpoint is mountpoint\n    mountpoint \"%a\"\nelif [ \"${mtpmount##*/}\" = \"aft-mtp-mount\" ]; then\n    $mtpmount -f \"%a\"\nelse\n    $mtpmount \"%a\"\nfi\n",
         FALSE,
